cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -195,7 +212,47 @@ func UpgradeControlPlaneComponents(flags *applyFlags, client clientset.Interface
 	// OK, the cluster is hosted using static pods. Upgrade a static-pod hosted cluster
 	fmt.Printf("[upgrade/apply] Upgrading your Static Pod-hosted control plane to version %q...\n", flags.newK8sVersionStr)
 
-	if err := upgrade.StaticPodControlPlane(client, internalcfg, flags.newK8sVersion); err != nil {
+	if flags.dryRun {
+		return DryRunStaticPodUpgrade(internalcfg)
+	}
+	return PerformStaticPodUpgrade(client, waiter, internalcfg)
+}
+
+// PerformStaticPodUpgrade performs the upgrade of the control plane components for a static pod hosted cluster
+func PerformStaticPodUpgrade(client clientset.Interface, waiter apiclient.Waiter, internalcfg *kubeadmapi.MasterConfiguration) error {
+	pathManager, err := upgrade.NewKubeStaticPodPathManagerUsingTempDirs(constants.GetStaticPodDirectory())
+	if err != nil {
+		return err
+	}
+
+	if err := upgrade.PerformStaticPodControlPlaneUpgrade(waiter, pathManager, internalcfg); err != nil {
```

Comment:
The name again is weird.. 
upgrade.PerformStaticPodControlPlaneUpgrade

vs. upgrade.StaticPodControlPlane

Reply:
renamed to `StaticPodControlPlane`